py/desispec/util.py

```diff
@@ -83,45 +93,53 @@ def runcmd(cmd, args=None, inputs=[], outputs=[], clobber=False):
             for x in outputs:
                 print("   ", x)
 
-    #- run command
-    if isinstance(cmd, collections.abc.Callable):
-        if args is None:
-            args = []
-
-        try:
-            return cmd(*args)
-        except Exception as e:
-            import traceback
-            lines = traceback.format_exception(*sys.exc_info())
-            for line in lines:
-                line = line.strip()
-                log.error(f'{line}')
-            log.critical(f"FAILED {cmd=} with {args=}")
-            raise e
+    cmd_callable = isinstance(cmd, collections.abc.Callable)
+    if not cmd_callable and args is not None:
+        raise ValueError("Don't provide args unless cmd is a function")
+    if isinstance(args, argparse.Namespace):
+        args = (args)
```

Comment:
This should be `args = (args,)` to convert to a tuple.  Otherwise it is just interpreted as parentheses and remains a Namespace.  Note that `blat = [blat]` works to convert to a list, but `blat = (blat)` doesn't work to convert to a tuple.  Stylistically I prefer to keep the extra comma in both cases to be clearer (?) that you really mean for this to become a list or tuple.

Reply:
thanks for finding this, I added the comma